src/python/pants/testutil/rule_runner.py

```diff
@@ -462,13 +462,24 @@ def get(product, subject):
 @contextmanager
 def mock_console(
     options_bootstrapper: OptionsBootstrapper,
+    *,
+    stdin_content: bytes | str | None = None,
 ) -> Iterator[Tuple[Console, StdioReader]]:
     global_bootstrap_options = options_bootstrapper.bootstrap_options.for_global_scope()
-    with initialize_stdio(global_bootstrap_options), open(
-        "/dev/null", "r"
-    ) as stdin, temporary_file(binary_mode=False) as stdout, temporary_file(
+
+    @contextmanager
+    def stdin_context():
+        if stdin_content is None:
+            yield open("/dev/null", "r")
+        else:
+            with temporary_file(binary_mode=isinstance(stdin_content, bytes)) as stdin_file:
```

Comment:
Can this just be `yield io.StringIO(stdin_content)` with a tightening of the `stdin_content` type to `str | None`?

Reply:
No, because it won't have a filehandle: `stdio_destination` needs a file handle to give to Rust.